src/Network-Kernel/Socket.class.st

```diff
@@ -1639,18 +1630,14 @@ Socket >> waitForDataFor: timeout [
 Socket >> waitForDataFor: timeout ifClosed: closedBlock ifTimedOut: timedOutBlock [
 	"Wait for the given nr of seconds for data to arrive."
 
-	| startTime msecsDelta |
+	| startTime msecsDelta msecsElapsed |
 	startTime := Time millisecondClockValue.
 	msecsDelta := (timeout * 1000) truncated.
-	[ (Time millisecondsSince: startTime) < msecsDelta ]
-		whileTrue: [ self dataAvailable ifTrue: [ ^ self ].
-			self isConnected ifFalse: [ ^ closedBlock value ].
-			self readSemaphore waitTimeoutMilliseconds: (msecsDelta - (Time millisecondsSince: startTime) max: 0) ].
-
-	self dataAvailable
-		ifFalse: [ ^ (self isConnected
-				ifTrue: [ timedOutBlock ]
-				ifFalse: [ closedBlock ]) value ]
+	[ self dataAvailable ] whileFalse: [
+		self isConnected ifFalse: [ ^ closedBlock value ].
+		(msecsElapsed := Time millisecondsSince: startTime) < msecsDelta
+			ifFalse: [ ^ timedOutBlock value ].
+		self readSemaphore waitTimeoutMilliseconds:: msecsDelta - msecsElapsed ]
```

Comment:
Here there is a typo (two colons :: ) and it should be fixed so we can integrate it.

Reply:
D'oh! Bad cherry-pick—I made these changes in an image where that method was still `waitTimeoutMSecs:` and Git did an _unusually_ bad job of detecting _what_ the conflict actually was so I had to do a lot of manual editing. Fixed now.